bastion/task/cmd/story.go

```diff
@@ -0,0 +1,383 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to process second column: %v", err)
+	}
+
+	if type1 == model.StoryMediaType.Image && type2 == model.StoryMediaType.Text {
+		imageSection = &model.StorySectionModel{
+			Type: model.StorySectionType.ImageText,
+			Data: []*model.StoryMediaModel{
+				{
+					Type: type1,
+					Url:  contentOrUrl1,
+				},
+				{
+					Type:    type2,
+					Content: contentOrUrl2,
+				},
+			},
+		}
+	} else if type1 == model.StoryMediaType.Text && type2 == model.StoryMediaType.Image {
+		imageSection = &model.StorySectionModel{
+			Type: model.StorySectionType.TextImage,
+			Data: []*model.StoryMediaModel{
+				{
+					Type:    type1,
+					Content: contentOrUrl1,
+				},
+				{
+					Type: type2,
+					Url:  contentOrUrl2,
+				},
+			},
+		}
+	} else {
+		return nil, fmt.Errorf("invalid column type combination: %s %s", type1, type2)
+	}
+
+	return imageSection, nil
+}
+
+func processColumn(block *notion.Block) (string, string, error) {
+	if !block.HasChildren || block.Type != notion.BlockTypes.Column {
+		return "", "", fmt.Errorf("invalid column block, id: %s", block.Id)
+	}
+
+	blockReponse, err := notionClient.RetrieveBlockChildren(block.Id, taskCfg.NotionAuthToken)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to get children of column block: %v", err)
+	}
+	if len(blockReponse.Results) == 0 {
+		return "", "", fmt.Errorf("invalid column block children response, children == 0")
+	}
+
+	var mediaType string
+	var contentOrUrl string
+	blockType := blockReponse.Results[0].Type
+	switch blockType {
+
+	case notion.BlockTypes.Image:
+		mediaType = model.StoryMediaType.Image
+		imageBlock := blockReponse.Results[0]
+		if imageBlock.Image == nil || imageBlock.Image.File == nil || imageBlock.Image.File.Url == "" {
+			return "", "", fmt.Errorf("invalid image block: id %s", imageBlock.Id)
+		}
+
+		contentOrUrl, err = transferImage(imageBlock.Image.File.Url)
+		if err != nil {
+			return "", "", fmt.Errorf("failed to transfer image block %s: %v", imageBlock.Id, err)
+		}
+
+	case notion.BlockTypes.Paragraph:
+		mediaType = model.StoryMediaType.Text
+		contentOrUrl, err = transformParagraphToText(blockReponse.Results)
+		if err != nil {
+			return "", "", fmt.Errorf("failed to transform paragraph to text %s: %v", block.Id, err)
+		}
+
+	default:
+		return "", "", fmt.Errorf("invalid block type within the column: id %s", block.Id)
+	}
+
+	return mediaType, contentOrUrl, nil
+}
+
+func transformParagraphToText(blocks []*notion.Block) (string, error) {
+	var text string
+	for idx, block := range blocks {
+		if block.Type != notion.BlockTypes.Paragraph {
+			return "", fmt.Errorf("invalid paragraph block %s", block.Id)
+		}
+		newContent, err := getContent(block.Paragraph)
+		if err != nil {
+			return "", fmt.Errorf("failed to get content for block %d, id %s: %v", idx, block.Id, err)
+		}
+		text = text + newContent
+	}
+	return text, nil
+}
+
+func init() {
```

Comment:
maybe you want to support multiple input formats later. please consider to use a separate module to support the notion page right now.